File: app/src/main/java/com/example/tipper/MainActivity.java
package com.example.tipper;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText; // for bill amount input
import android.widget.SeekBar; // for changing the tip percentage
import android.widget.SeekBar.OnSeekBarChangeListener; // SeekBar listener
import android.widget.TextView; // for displaying text

import java.text.NumberFormat; // for currency formatting

public class MainActivity extends AppCompatActivity {

    private Button _btnBMICalculator;
    private Button _btnCaloriesCounter;
    private Button _btnQuiz;
    private Button _btnCannonGame;
    // called when the activity is first created
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState); // call superclass onCreate
        setContentView(R.layout.activity_main); // inflate the GUI

        _btnBMICalculator = (Button) findViewById(R.id.btnBMICalc);
        _btnCaloriesCounter = (Button) findViewById(R.id.btnCaloriesCounter);
        _btnQuiz = (Button) findViewById(R.id.btnQuiz);
        _btnCannonGame = (Button) findViewById(R.id.btnCannonGame);

        _btnBMICalculator.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view){
                GoToBMICalculator();
            }
        });

        _btnCaloriesCounter.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view){
                GoToCaloriesCounter();
            }
        });

        _btnQuiz.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view){
                GoToQuiz();
            }
        });
        _btnCannonGame.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view){
                GoToCannonGame();
            }
        });
    }

    private void GoToBMICalculator(){
        Intent intent = new Intent(this, BMICalculator.class);
        startActivity(intent);
    }

    private void GoToCaloriesCounter(){
        Intent intent = new Intent(this, CaloriesCounter.class);
        startActivity(intent);
    }

    private void GoToQuiz(){
        Intent intent = new Intent(this, QuizActivity.class);
        startActivity(intent);
    }

    private void GoToCannonGame(){
        Intent intent = new Intent(this, CannonActivity.class);
        startActivity(intent);
    }
}



/*************************************************************************
 * (C) Copyright 1992-2016 by Deitel & Associates, Inc. and               *
 * Pearson Education, Inc. All Rights Reserved.                           *
 *                                                                        *
 * DISCLAIMER: The authors and publisher of this book have used their     *
 * best efforts in preparing the book. These efforts include the          *
 * development, research, and testing of the theories and programs        *
 * to determine their effectiveness. The authors and publisher make       *
 * no warranty of any kind, expressed or implied, with regard to these    *
 * programs or to the documentation contained in these books. The authors *
 * and publisher shall not be liable in any event for incidental or       *
 * consequential damages in connection with, or arising out of, the       *
 * furnishing, performance, or use of these programs.                     *
 *************************************************************************/

File: app/src/main/java/com/example/tipper/MainActivity.kt
package com.example.tipper

import org.junit.runner.RunWith
import com.example.tipper.CannonView
import com.example.tipper.Cannonball
import com.example.tipper.GameElement
import com.example.tipper.CannonView.CannonThread
import android.app.Activity
import com.example.tipper.Cannon
import com.example.tipper.Blocker
import android.media.SoundPool
import android.util.SparseIntArray
import com.example.tipper.R
import android.os.Build
import android.media.AudioAttributes
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.content.Intent
import com.example.tipper.BMICalculator
import com.example.tipper.CaloriesCounter
import com.example.tipper.QuizActivity
import com.example.tipper.CannonActivity
import com.example.tipper.QuizController
import android.widget.TextView
import android.widget.EditText
import android.widget.CheckBox
import com.example.tipper.QuizQuestion
import com.example.tipper.RecipesActivity
import android.media.AudioManager
import android.view.*
import android.widget.Button

// for bill amount input
// for changing the tip percentage
// SeekBar listener
// for displaying text
// for currency formatting
class MainActivity : AppCompatActivity() {
    private var _btnBMICalculator: Button? = null
    private var _btnCaloriesCounter: Button? = null
    private var _btnQuiz: Button? = null
    private var _btnCannonGame: Button? = null

    // called when the activity is first created
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState) // call superclass onCreate
        setContentView(R.layout.activity_main) // inflate the GUI
        _btnBMICalculator = findViewById<View?>(R.id.btnBMICalc) as Button
        _btnCaloriesCounter = findViewById<View?>(R.id.btnCaloriesCounter) as Button
        _btnQuiz = findViewById<View?>(R.id.btnQuiz) as Button
        _btnCannonGame = findViewById<View?>(R.id.btnCannonGame) as Button
        _btnBMICalculator!!.setOnClickListener(View.OnClickListener { GoToBMICalculator() })
        _btnCaloriesCounter!!.setOnClickListener(View.OnClickListener { GoToCaloriesCounter() })
        _btnQuiz!!.setOnClickListener(View.OnClickListener { GoToQuiz() })
        _btnCannonGame!!.setOnClickListener(View.OnClickListener { GoToCannonGame() })
    }

    private fun GoToBMICalculator() {
        val intent = Intent(this, BMICalculator::class.java)
        startActivity(intent)
    }

    private fun GoToCaloriesCounter() {
        val intent = Intent(this, CaloriesCounter::class.java)
        startActivity(intent)
    }

    private fun GoToQuiz() {
        val intent = Intent(this, QuizActivity::class.java)
        startActivity(intent)
    }

    private fun GoToCannonGame() {
        val intent = Intent(this, CannonActivity::class.java)
        startActivity(intent)
    }
}